articles/openshift/howto-deploy-java-eap-app.md

```diff
@@ -0,0 +1,384 @@
+
+   Though the "Get a Red Hat pull secret" step is labeled as optional, **it is required for this article**.  The pull secret enables your Azure Red Hat OpenShift cluster to find the JBoss EAP application images.
+
+   If you plan to run memory-intensive applications on the cluster, specify the proper virtual machine size for the worker nodes using the `--worker-vm-size` parameter. For more information, see:
+
+   * [Azure CLI to create a cluster](/cli/azure/aro#az_aro_create)
+   * [Supported virtual machine sizes for memory optimized](./support-policies-v4.md#memory-optimized)
+
+1. Connect to the cluster by following the steps in [Connect to an Azure Red Hat OpenShift 4 cluster](./tutorial-connect-cluster.md).
+   * Follow the steps in "Install the OpenShift CLI" 
+   * Connect to an Azure Red Hat OpenShift cluster using the OpenShift CLI with the user `kubeadmin`
+
+1. Execute the following command to create the OpenShift project for this demo application:
+
+    ```bash
+    $ oc new-project eap-demo
+    Now using project "eap-demo" on server "https://api.zhbq0jig.northeurope.aroapp.io:6443".
+
+    You can add applications to this project with the 'new-app' command. For example, try:
+
+    oc new-app rails-postgresql-example
+
+    to build a new example application in Ruby. Or use kubectl to deploy a simple Kubernetes application:
+
+    kubectl create deployment hello-node --image=k8s.gcr.io/serve_hostname
+    ```
+
+1. Execute the following command to add the view role to the default service account. This role is needed so the application can discover other pods and form a cluster with them:
+
+    ```bash
+    $ oc policy add-role-to-user view system:serviceaccount:$(oc project -q):default -n $(oc project -q)
+    clusterrole.rbac.authorization.k8s.io/view added: "system:serviceaccount:eap-demo:default"
+    ```
+
+## Prepare the application
+
+At this stage, you have cloned the demo application and your local repository is on the `main` branch. The demo application is a simple Jakarta EE 8 application that creates, reads, updates, and deletes records on a Microsoft SQL Server. Usually, you could have a JBoss EAP server installed on your local machine and configured with the required drivers and data source to allow connections from your application to the database server available on your environment.
+
+When you are targeting OpenShift, you might want to trim the capabilities of your JBoss EAP server, for example, to reduce the security exposure of the provisioned server and reduce the overall footprint. You could also want to include some Microprofile specs to make your application more suitable for running on an OpenShift environment. When using JBoss EAP, one way to accomplish this is by packaging your application and your server in a single deployment unit known as Bootable JAR. Let's do that by adding the required changes to our demo application.
```

Comment:
I am not sure how including MicroProfile specs would make an application more suitable for running on an OpenShift environment?